features/dd-sdk-android-trace/src/main/kotlin/com/datadog/android/trace/internal/DatadogTracingToolkit.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+package com.datadog.android.trace.internal
+
+import androidx.annotation.IntRange
+import com.datadog.android.lint.InternalApi
+import com.datadog.android.trace.api.span.DatadogSpanContext
+import com.datadog.android.trace.api.tracer.DatadogTracerBuilder
+
+/**
+ * For library usage only.
+ * Provides implementation for specific interfaces to dependent modules
+ */
+@InternalApi
+object DatadogTracingToolkit {
+    /**
+     * Provides a mechanism for converting Datadog span IDs between decimal and hexadecimal representations.
+     *
+     * This converter is utilized to ensure span ID consistency and proper formatting for distributed tracing
+     * when working with the Datadog SDK.
+     */
+    @JvmField
+    val spanIdConverter: DatadogSpanIdConverter = DatadogSpanIdConverter()
+
+    /**
+     * Providing helper function to extract [DatadogSpanContext] from tracing context added by addParentSpan method.
+
+     * This property is intended for internal usage only and should not
+     * be altered externally.
+     */
+    var propagationHelper: DatadogPropagationHelper = DatadogPropagationHelper()
+        internal set
+
+    internal var builderProvider: DatadogTracerBuilder? = null
```

Comment:
let's call it `testBuilderProvider`? it is set only in tests in seems.

I'm not quite a fan of having test references in the production code, but on the opposite side is mocking static `newTracerBuilder` - maybe we can create a TODO item to somehow improve such setup in the future.